### ISSUE TITLE
avoid_shadowing_type_parameters doesn't work in extensions

### ISSUE THREAD

**Issue by natebosch** [dart-lang/linter#2033]
Expected a lint in:

```dart
extension A<T> on List<T> {
  void fn<T>() {}
}
```

See https://github.com/dart-lang/language/issues/885

**Comment by pq** [comment]
Good catch.  Thanks!

**Comment by pq** [comment]
(Aside: looks like mixins are skipped too 😬.)
